virtcontainers/sandbox.go

```diff
@@ -1923,6 +1923,14 @@ func (s *Sandbox) calculateSandboxMemory() int64 {
 
 		if m := c.Resources.Memory; m != nil && m.Limit != nil {
 			memorySandbox += *m.Limit
+			s.Logger().WithField("memory limit", memorySandbox).Info("Memory Sandbox + Memory Limit ")
+		}
+
+		//Add hugepages memory
+		//HugepageLimit is uint64 - https://github.com/opencontainers/runtime-spec/blob/master/specs-go/config.go#L242
+		for _, l := range c.Resources.HugepageLimits {
+			memorySandbox += int64(l.Limit)
```

Comment:
This is confusing: is it `unit64` or `int64`?

Reply:
the HugepageLimit is in uint64 :-(

Reply:
Hmm. I guess you should check that `l.Limit <= math.MaxInt64` herethen. If it's bigger, set it to `math.MaxInt64` and make a logger call (but ideally update the function to return an error).

Reply:
You are right. Sorry for the missing this.   Do you recommend returning error (which will change the current function) or just make a logger call ?

Reply:
I'd make `calculateSandboxMemory()` return an `error`. It's only called from one other function (`updateResources()`) and that function already returns an `error` so not too disruptive to update it. Also, by returning an `error` you could add a unit test to exercise the `EOVERFLOW` scenario.

Reply:
sure.. I'll make the changes